### PR TITLE
Unbuckling Closes Inventory UI

### PR DESCRIPTION
**Pull request by MehimoNemo**
What a fucking mudcrawl this was.
Fixes a line of code that caused parented objects to receive incorrect positions, causing, apparently, all sorts of icky stuff.
My brain is hot mush.

### CODE REVIEW

File: Robust.Shared/GameObjects/Components/Transform/TransformComponent.cs [hunk 1]
```diff
@@ -542,6 +542,10 @@ public virtual void AttachParent(ITransformComponent newParent)
                 return;
             }
 
+
+            // offset position from world to parent
+            SetPosition(WorldPosition - newParent.WorldPosition);
```
Comment (ShadowCommander) [inline Robust.Shared/GameObjects/Components/Transform/TransformComponent.cs]: Move this to just above `_parent = newParentEnt.Uid;` on line 556.

File: Robust.Shared/GameObjects/Components/Transform/TransformComponent.cs [hunk 2]
```diff
@@ -560,8 +564,7 @@ public virtual void AttachParent(ITransformComponent newParent)
             Owner.EntityManager.EventBus.RaiseLocalEvent(Owner.Uid, entMessage);
             Owner.SendMessage(this, compMessage);
 
-            // offset position from world to parent
-            SetPosition(newParent.InvWorldMatrix.Transform(_localPosition));
+
```
Comment (ShadowCommander) [inline Robust.Shared/GameObjects/Components/Transform/TransformComponent.cs:564]: Remove the unnecessary blank lines so we have clean formatting.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

**Comment by MehimoNemo** [comment]
Okay, I think that fixes it? Probably

**Review by ShadowCommander — APPROVED** [review]

**Comment by MehimoNemo** [comment]
Not sure if this is all green, UI bug still occurring???

**Comment by ShadowCommander** [comment]
> Not sure if this is all green, UI bug still occurring???

The bug will still be there until we update the RobustToolbox submodule in the space-station-14 repo, so don't worry. It's fixed, the server just hasn't been updated.

**Pull request merged**